cpp/tests/sort/sort_test.cpp

```diff
@@ -197,6 +197,133 @@ TYPED_TEST(Sort, WithAllValid)
   }
 }
 
+TYPED_TEST(Sort, WithStructColumn)
```

Comment:
Copyright year in the header needs to be updated. Also, please test for the case of sliced columns---this is very important.